components/src/hardware-sim/RobotCoordinateSpace/RobotCoordinateSpaceWithDOMCoords.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react'
+import { Svg } from '../../primitives'
+import type { DeckDefinition, DeckSlot } from '@opentrons/shared-data'
+
+export interface RobotCoordinateSpaceWithDOMCoordsRenderProps {
+  deckSlotsById: { [slotId: string]: DeckSlot }
+  getRobotCoordsFromDOMCoords: (
+    domX: number,
+    domY: number
+  ) => { x: number; y: number }
+}
+
+interface RobotCoordinateSpaceWithDOMCoordsProps
+  extends React.ComponentProps<typeof Svg> {
+  viewBox?: string | null
+  deckDef?: DeckDefinition
+  children?: (
+    props: RobotCoordinateSpaceWithDOMCoordsRenderProps
+  ) => React.ReactNode
+}
+
+type GetRobotCoordsFromDOMCoords = RobotCoordinateSpaceWithDOMCoordsRenderProps['getRobotCoordsFromDOMCoords']
+
+export function RobotCoordinateSpaceWithDOMCoords(
+  props: RobotCoordinateSpaceWithDOMCoordsProps
+): JSX.Element | null {
+  const { children, deckDef, viewBox, ...restProps } = props
+  const wrapperRef = React.useRef<SVGSVGElement>(null)
+  const getRobotCoordsFromDOMCoords: GetRobotCoordsFromDOMCoords = (x, y) => {
+    if (!wrapperRef.current) return { x: 0, y: 0 }
```

Comment:
I guess doing null check might be general 🤔 if there is no strong reasons to do use `!wrapperRef.current`.

```suggestion
    if (wrapperRef.current == null) return { x: 0, y: 0 }
```

Reply:
ya i basically copied and pasted that logic from `RobotWorkSpace`, so the logic is a bit dated, I'll update it to your suggestion. I think it makes more sense to check for if it isn't null instead.